jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3VirtualBucket.kt

```diff
@@ -49,6 +52,18 @@ class S3VirtualBucket(val s3Bucket: Bucket, val client: S3Client) : LightVirtual
         }
     }
 
+    suspend fun getBucketVersioning(): GetBucketVersioningResponse = withContext(Dispatchers.IO) {
+        client.getBucketVersioning {
+            it.bucket(s3Bucket.name())
+        }
+    }
+
+    suspend fun listVersionObjects(key: String): ListObjectVersionsResponse = withContext(Dispatchers.IO) {
+        client.listObjectVersions {
```

Comment:
This call can be paginated. `listObjectVersionsPaginator` feels acceptable to me here, but might be worth considering adopting the continuation pattern used in the regular `listObjects` now, or as a TODO

Reply:
Changed to `listObjectVersionsPaginator` and in the end I've moved continuation pattern logic into new `S3LazyLoadParentNode` class so that `S3TreeDirectoryNode` and `S3TreeObjectNode` only define how to `loadObjects`